test/IRGen/builtins.swift

```diff
@@ -804,7 +804,15 @@ enum MyError : Error {
 
 throw MyError.A
 
+/// Builtin.globalStringTablePointer must be reduced to a string_literal instruction before IRGen. IRGen
+/// should make this a trap.
+// CHECK-LABEL: define {{.*}}globalStringTablePointer
+// CHECK: call void @llvm.trap()
+// CHECK: ret i8* undef
+func globalStringTablePointerUse(_ str: String) -> Builtin.RawPointer {
+  return Builtin.globalStringTablePointer(str);
+}
```

Comment:
Should it compile to a trap, or fail to compile?

Reply:
It compiles to a trap to enable them to wrapped inside other `@_transparent` functions,  and eventually some client function (possibly form a different module) define the constant string.

E.g. the SPI `_globalStringTablePointer` in `Builtin.swift` wraps this builtin and applies it to a non-constant argument. This SPI must compile as a part of stdlib.  Also, the SPI itself is used the `osLog` function which still applies the builtin to a non-constant string obtained from a parameter. Eventually, the constant is produced only when the client of `osLog` is seen.  Also each of these functions: `_globalStringTablePointer`, `osLog` are compiled into separate modules and so the calling context is not available when they are compiled.

Reply:
May be, we can emit a diagnostics if the builtin is applied to a non-literal string in a non-transparent function. However, we need the IRGen to emit a trap for transparent functions.

Reply:
Btw, based on my discussions,  if `@_alwaysEmitIntoClient` functions are IRGen'ed lazily like private methods of classes, it would be possible to make the IRGen for this builtin: `Bultin.globalStringTablePointer` as `llvm::unreachable`. However, as of now, `@_alwaysEmitIntoClient` will pass through IRGen whether or not they are used within the module.